### DISCUSSION TITLE
Saving data to localStorage

### DISCUSSION THREAD

**Discussion by PaTp0H** [react-hook-form/react-hook-form#4292] [Q&A]
Hi guys!
I am doing the task of saving user data to local storage, and I have a problem, please help me to solve it)

I am monitoring user input using the watch () method and I get the error `Warning: Maximum update depth exceeded. This can happen when a component calls setState inside useEffect, but useEffect either doesn't have a dependency array, or one of the dependencies changes on every render. I run out of ideas how to fix it, the implementation code is below:
  useEffect(() => {
    if (isDirty) {
      setStorageData(getValues());
    }
  }, [watch()]);

setStorageData was taken from here and slightly modified, but the logic remains the same https://usehooks-typescript.com/react-hook/use-local-storage

**Comment by bluebill1049** [comment]
```
const value = watch();

isDirty  && setStorageData(value);
```

try above.
